index.ts

```diff
@@ -0,0 +1,476 @@
+	protect?: MoveFlag; // Blocked by Detect, Protect, Spiky Shield, and if not a Status move, King's Shield.
+	pulse?: MoveFlag; // Power is multiplied by 1.5 when used by a Pokemon with the Ability Mega Launcher.
+	punch?: MoveFlag; // Power is multiplied by 1.2 when used by a Pokemon with the Ability Iron Fist.
+	recharge?: MoveFlag; // If this move is successful, the user must recharge on the following turn and cannot make a move.
+	reflectable?: MoveFlag; // Bounced back to the original user by Magic Coat or the Ability Magic Bounce.
+	snatch?: MoveFlag; // Can be stolen from the original user and instead used by another Pokemon using Snatch.
+	sound?: MoveFlag; // Has no effect on Pokemon with the Ability Soundproof.
+}
+
+type MoveCategory: 'Physical' | 'Special' | 'Status';
+
+type MoveTarget =
+	// single-target
+	'normal'|'any'|'adjacentAlly'|'adjacentFoe'|'adjacentAllyOrSelf'|
+	// single-target, automatic
+	'self'|'randomNormal'|
+	// spread
+	'allAdjacent'|'allAdjacentFoes'|
+	// side and field
+	'allySide'|'foeSide'|'all';
+
+interface SelfEffect {
+	volatileStatus?: string;
+	boosts?: Partial<BoostsTable>;
+}
+
+interface SecondaryEffect {
+	chance?: number;
+	boosts?: Partial<BoostsTable>;
+	self?: SelfEffect;
+	status?: StatusName;
+	volatileStatus?: string;
+
+}
+interface Move extends Data {
+	kind: 'Move';
+
+	accuracy: number | true;
+	basePower: number;
+	flags: MoveFlags;
+	pp: number;
+	target: MoveTarget;
+	type: TypeName;
+
+	alwaysCrit?: boolean;
+	alwaysHit?: boolean;
+	basePowerModifier?: number;
+	boosts?: Partial<BoostsTable> | false
+	bypassesProtect?: boolean;
+	category: MoveCategory;
+	contestType?: string;
+	critModifier?: number;
+	critRatio?: number;
+	damage?: number | 'level' | false | null;
+	defensiveCategory?: MoveCategory;
+	drain?: [number, number];
+	forceSTAB?: boolean;
+	forceSwitch?: boolean;
+	givesHealth?: boolean;
+	hasCustomRecoil?: boolean;
+	hasPriority?: boolean;
+	hasRecoil?: Recoil;
+	hasSecondaryEffect?: boolean;
+	heal?: number[] | null;
+	ignoreAbility?: boolean;
+	ignoreAccuracy?: boolean;
+	ignoreDefensive?: boolean;
+	ignoreEvasion?: boolean;
+	ignoreImmunity?: boolean | {[k: string]: boolean};
+	ignoreNegativeOffensive?: boolean;
+	ignoreOffensive?: boolean;
+	ignorePositiveDefensive?: boolean;
+	ignorePositiveEvasion?: boolean;
+	ignoresBurn?: boolean;
+	ignoresDefenseBoosts?: boolean;
+	infiltrates?: boolean;
+	isFutureMove?: boolean;
+	isSpread?: boolean | 'allAdjacent';
+	isZ?: boolean | ID;
+	multiaccuracy?: boolean;
+	multihit?: number | number[];
+	multihitType?: 'parentalbond';
+	noCopy?: boolean;
+	noDamageVariance?: boolean;
+	noFaint?: boolean;
+	noMetronome?: string[];
+	noPPBoosts?: boolean;
+	noSketch?: boolean;
+	nonGhostTarget?: string;
+	ohko?: true | 'Ice' | null;
+	percentHealed?: number;
+	pressureTarget?: string;
+	priority?: number;
+	pseudoWeather?: string:
+	recoil?: [number, number];
```

Comment:
would like the fields to be named, don't know what this means

Reply:
I believe this is a fraction, like [1, 4] = 1/4

Reply:
i would multiply all fractions by `100`, so `[1, 4] = 25`. this is consistent with how accuracy is handled

Reply:
Per discussion in PM, I believe this may be required for accuracy reasons like with gen 4 woodhammer recoil being `[1, 3]`. @Marty

Reply:
I'd be fine with `33` and a hardcoded "handle this differently in Gen 4".

We also use arrays for fractions in `heal`, and for ranges in `multihit`.

Reply:
Yeah, 33 sounds good but won't be handled as 33/100 in Gen 5+ either, it's really some value out of 4096 I can't remember right now. I'm good with swapping fractions like this. (It's more consistent with the move data in the games too!)